app/src/interfaces/map/map.vue

```diff
@@ -1,48 +1,76 @@
 <template>
 	<div class="interface-map">
-		<div class="map" ref="container"></div>
-		<div v-if="!canRenderBackground" class="error">
-			<v-info icon="vpn_key" center type="warning" :title="$t('interfaces.map.no_api_key')"></v-info>
+		<div class="map" ref="container" :class="{ error }"></div>
+		<div v-if="error" class="info">
+			<v-info v-if="geometryParsingError" icon="error" center type="warning" :title="$t('incompatible_geometry')">
+				<template #append>
+					<v-button small @click="resetInterface" class="reset-preset">{{ $t('reset_interface') }}</v-button>
+				</template>
+			</v-info>
 		</div>
 	</div>
 </template>
 
 <script lang="ts">
-import { computed, defineComponent, onMounted, PropType, ref, watch, watchEffect } from '@vue/composition-api';
-import 'maplibre-gl/dist/maplibre-gl.css';
-
+import {
+	computed,
+	defineComponent,
+	onMounted,
+	onUnmounted,
+	PropType,
+	ref,
+	toRefs,
+	watch,
+	watchEffect,
+} from '@vue/composition-api';
 import maplibre, {
 	MapLayerMouseEvent,
 	LngLatBoundsLike,
 	GeoJSONSource,
+	AnimationOptions,
 	CameraOptions,
 	LngLatLike,
 	Source,
-	Point,
 	Style,
 	Map,
 	Marker,
 	LngLat,
+	IControl,
 } from 'maplibre-gl';
-import { ButtonControl } from '@/layouts/map/components/map.vue';
-import { sources, mapbox_sources } from '@/layouts/map/styles/sources';
-import { Point as JSONPoint } from 'geojson';
-import proj4 from 'proj4';
-import { useSettingsStore } from '@/stores';
-import getSetting from '@/utils/get-setting';
-
-type ModelValue = JSONPoint | [string, string] | null;
+import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
+import MapboxDraw from '@mapbox/mapbox-gl-draw';
+import { Position, Point, Polygon, LineString, MultiPoint, MultiPolygon, MultiLineString, BBox } from 'geojson';
+import { getBasemapSources, BasemapSource, getStyleFromBasemapSource } from '@/layouts/map/basemap';
+import { ControlButton, ControlGroup } from '@/layouts/map/controls';
+import { getParser, getSerializer, assignBBox } from '@/layouts/map/lib';
+import { GeometryFormat, AnyGeoJSON } from '@/layouts/map/lib';
```

Comment:
Lets figure out if we can put this in a shared component folder instead of having the interface pull from the layout 🤔 (Trying to keep them as 'scoped' as possible)

Reply:
Totally agree.